PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/link/mysql/PasswordDialog.java

```diff
@@ -0,0 +1,157 @@
+import com.microsoft.azure.toolkit.intellij.common.AzureDialog;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.form.AzureForm;
+import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTask;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import com.microsoft.azuretools.azurecommons.util.Utils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class PasswordDialog extends AzureDialog<PasswordConfig> implements AzureForm<PasswordConfig> {
+
+    private static final String TITLE = "Credential for Azure Database for MySQL";
+    private static final String HEADER_PATTERN = "Please provide credential for user (%s) to access database (%s) on server (%s).";
+
+    private JPanel root;
+    private JLabel headerIconLabel;
+    private JTextPane headerTextPane;
+    private JTextPane testResultTextPane;
+    private JLabel testResultLabel;
+    private JButton testConnectionButton;
+    private JButton testResultButton;
+    private JPasswordField passwordField;
+    private PasswordSaveComboBox passwordSaveComboBox;
+
+    private Project project;
+    private String username;
+    private String url;
+
+    public PasswordDialog(Project project, String username, String url) {
+        super(project);
+        this.project = project;
+        this.username = username;
+        this.url = url;
+        setTitle(TITLE);
+        JdbcUrl jdbcUrl = JdbcUrl.from(url);
+        headerTextPane.setText(String.format(HEADER_PATTERN, username, jdbcUrl.getDatabase(), jdbcUrl.getHostname()));
+        testConnectionButton.setEnabled(false);
+        testResultLabel.setVisible(false);
+        testResultButton.setVisible(false);
+        testResultTextPane.setEditable(false);
+        testResultTextPane.setText(StringUtils.EMPTY);
+        Dimension lastColumnSize = new Dimension(106, 38);
+        passwordSaveComboBox.setPreferredSize(lastColumnSize);
+        passwordSaveComboBox.setMaximumSize(lastColumnSize);
+        passwordSaveComboBox.setSize(lastColumnSize);
+        this.init();
+        this.initListener();
+    }
+
+    private void initListener() {
+        this.passwordField.addKeyListener(this.onInputPasswordFieldChanged());
+        this.testConnectionButton.addActionListener(this::onTestConnectionButtonClicked);
+        this.testResultButton.addActionListener(this::onCopyButtonClicked);
+
+    }
+
+    private KeyListener onInputPasswordFieldChanged() {
```

Comment:
the name of this method is weird.

Reply:
onInputPasswordFieldChanged is  closer to UI than onPasswordChanged. isn't it? and all these methods of listeners follow this rule in this class.